src/caselawclient/models/identifiers/collection.py

```diff
@@ -22,20 +21,54 @@
 
 
 class IdentifiersCollection(dict[str, Identifier]):
-    def validate_uuids_match_keys(self) -> None:
+    def validate_uuids_match_keys(self) -> SuccessFailureMessageTuple:
         for uuid, identifier in self.items():
             if uuid != identifier.uuid:
-                msg = "Key of {identifier} in Identifiers is {uuid} not {identifier.uuid}"
-                raise UUIDMismatchError(msg)
+                return SuccessFailureMessageTuple(
+                    False, [f"Key of {identifier} in Identifiers is {uuid} not {identifier.uuid}"]
+                )
+
+        return SuccessFailureMessageTuple(True, [])
+
+    def check_only_single_current_identifier_per_type(self) -> SuccessFailureMessageTuple:
```

Comment:
This will probably want tweaking when we might have multiple identifiers per document; is it worth changing the name to reflect this to avoid API changes later?

`check_no_multiple_identifiers_for_solo_identifier_types`?